src/core_modules/capture-core/components/DataEntry/dataEntryField/internal/dataEntryField.utils.js

```diff
@@ -1,28 +1,28 @@
 // @flow
 import i18n from '@dhis2/d2-i18n';
 
-type Validator = (value: any) => boolean | { valid: boolean, message: ?string };
+type Validator = (value: any, internalComponentError?: ?Object) => boolean | { valid: boolean, errorMessage?: string };
 
 export type ValidatorContainer = {
     validator: Validator,
     message: string,
 };
 
-export function getValidationError(value: any, validatorContainers: ?Array<ValidatorContainer>) {
+export function getValidationError(value: any, validatorContainers: ?Array<ValidatorContainer>, internalComponentError?: ?Object) {
     if (!validatorContainers) {
         return null;
     }
 
     let message;
     const errorEncountered = validatorContainers.some((validatorContainer) => {
         const validator = validatorContainer.validator;
-        const result = validator(value);
+        const result = validator(value, internalComponentError);
 
         if (result === true || (result && result.valid)) {
             return false;
         }
 
-        message = (result && result.message) || validatorContainer.message;
+        message = (result && result.errorMessage) || validatorContainer.message;
```

Comment:
I really think we shouldn't change this from `message` to `errorMessage` as this is being used by a lot of validators (I looked up one of the validators and it was still using `message`). Let's make this easy on ourselves and keep using `message`. For example here we are using message https://github.com/dhis2/capture-app/blob/48713020465d8764a1ea3104a8d873216b9157d6/src/core_modules/capture-core/components/DataEntries/Enrollment/fieldValidators/enrollmentDate.validatorContainersGetter.js#L23 (this particular validator referenced here should also be rewritten to use our new validation rules)

Reply:
I modified this code to display the error message that is coming from calendarInput as what is returned from isValidDate is errorMessage not message.
I will change the code to be `(result && result.errorMessage) || (result && result.message) || validatorContainer.message;` so all scenarios work.